### ISSUE TITLE
index error messages

### ISSUE THREAD

**Issue by michaellilltokiwa** [tokiwa-software/fuzion#677]
Rust, GHC index their error messages, see: https://doc.rust-lang.org/error-index.html

Should we do the same?

**Comment by fridis** [comment]
Yes, that is in the back of my head since a while. Once we have this, the next step would be to have a set of tests, at least one for every error message.  This will also help us removing unused error messages.

**Comment by fridis** [comment]
What I really like is the way in the Eiffel language reference _Eiffel: The Language_ there are a number of validity rules defined. These rules have 4-letter ids that allow cross-referencing from errors to language spec rules. 
Ideally, the checks performed by the Fuzion compiler before producing an error become validity rules in the language specification. Both, error messages and validity rules should have an id and there should be a simple mapping between these two.
